gui/resultstree.cpp

```diff
@@ -288,9 +288,9 @@ void ResultsTree::setReportType(ReportType reportType) {
         if (!fileItem)
             continue;
         for (int j = 0; j < fileItem->rowCount(); ++j) {
-            const auto& data = fileItem->child(j,0)->data().toMap();
```

Comment:
Was there variable shadowing related to `data`? If so, could we detect that?

Reply:
```
D:\a\cppcheck\cppcheck\gui\resultstree.cpp(291,32): warning C4458: declaration of 'data' hides class member [D:\a\cppcheck\cppcheck\build\gui\cppcheck-gui.vcxproj]
D:\a\cppcheck\Qt\5.15.2\msvc2019_64\include\QtWidgets/qwidget.h(747,18): message : see declaration of 'QWidget::data' (compiling source file D:\a\cppcheck\cppcheck\gui\resultstree.cpp) [D:\a\cppcheck\cppcheck\build\gui\cppcheck-gui.vcxproj]
```

Not something we can detect until the system header is included. Also I think that compilers have this covered.

I just had a look at enabling `-Wshadow-field` for Clang but that warning has a wider scope.